### PR TITLE
Ex 3771

### PR DESCRIPTION
**Pull request by TWrightSovrn**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/sovrnBidAdapter.js [hunk 1]
```diff
@@ -74,22 +74,9 @@ export const spec = {
       let iv;
       let schain;
       let eids;
-      let tpid = []
       let criteoId;
 
       _each(bidReqs, function (bid) {
-        if (!eids && bid.userId) {
```
Comment (TWrightSovrn) [inline modules/sovrnBidAdapter.js]: This removes the TPIDS stuff.

File: modules/sovrnBidAdapter.js [hunk 2]
```diff
@@ -167,20 +154,29 @@ export const spec = {
...
         deepSetValue(sovrnBidReq, 'user.ext.eids', eids)
-        deepSetValue(sovrnBidReq, 'user.ext.tpid', tpid)
         if (criteoId) {
           deepSetValue(sovrnBidReq, 'user.ext.prebid_criteoid', criteoId)
         }
       }
 
-      let url = `https://ap.lijit.com/rtb/bid?src=$$REPO_AND_VERSION$$`;
-      if (iv) url += `&iv=${iv}`;
+      const urlParams = new URLSearchParams(window.location.search)
```
Comment (TWrightSovrn) [inline modules/sovrnBidAdapter.js]: This is only here for testing purposes using the integration tests stuff.

File: modules/sovrnBidAdapter.js [hunk 3]
```diff
@@ -268,11 +264,16 @@ export const spec = {
...
 function _buildVideoRequestObj(bid) {
   const videoObj = {}
+  const bidSizes = deepAccess(bid, 'sizes')
   const videoAdUnitParams = deepAccess(bid, 'mediaTypes.video', {})
   const videoBidderParams = deepAccess(bid, 'params.video', {})
   const computedParams = {}
-
-  if (Array.isArray(videoAdUnitParams.playerSize)) {
+  
+  if (bidSizes) {
```
Comment (TWrightSovrn) [inline modules/sovrnBidAdapter.js:285]: Changes wanted for size stuff, no clue if it is right.

**Review by yuffieh — APPROVED** [review]
lgtm

File: modules/sovrnBidAdapter.js [hunk 4]
```diff
@@ -85,7 +84,6 @@ export const spec = {
               if (id.source === 'criteo.com') {
                 criteoId = id.uids[0].id
               }
-              tpid.push({source: id.source, uid: id.uids[0].id})
```
Comment (TWrightSovrn) [inline modules/sovrnBidAdapter.js:88]: Removes tpids.

File: modules/sovrnBidAdapter.js [hunk 5]
```diff
@@ -167,7 +165,6 @@ export const spec = {
 
       if (eids) {
         deepSetValue(sovrnBidReq, 'user.ext.eids', eids)
-        deepSetValue(sovrnBidReq, 'user.ext.tpid', tpid)
```
Comment (TWrightSovrn) [inline modules/sovrnBidAdapter.js:170]: Also part of removing tpids.

File: test/spec/modules/sovrnBidAdapter_spec.js [hunk 6]
```diff
@@ -128,6 +128,42 @@ describe('sovrnBidAdapter', function() {
         expect(impression.video.startdelay).to.equal(startdelay)
       })
 
+      it('sets the proper video object wihtout sizes defined but video sizes defined', function() {
```
Comment (TWrightSovrn) [inline test/spec/modules/sovrnBidAdapter_spec.js:132]: Added to test if there was no sizes defined that it would use video sizes instead.

**Review by cpabst — APPROVED** [review]
LGTM

**Comment by TWrightSovrn** [comment]
Merged into prebid master

**Pull request closed without merging**